Allow custom username prefix while creating ephemeral Splunk user

abhide opened the pull request:
Tickets: TE-34

Commit:
Allow custom username prefix while creating ephemeral Splunk user

Tickets: TE-34

michaelw commented on README.md:
```diff
@@ -82,7 +93,7 @@ Create temporary admin account:
-    username           vault_local-admin_okta-mweber_70c6c140-238d-e12b-3289-8e38f8c4d9f5_1553712516020311000
+    username           vault_70c6c140-238d-e12b-3289-8e38f8c4d9f5
 
 This creates a new user account `vault_local-admin_okta-mweber_70c6...`
```
nit: update

abhide replied:
Done

michaelw commented on path_creds_create.go:
```diff
@@ -62,7 +60,7 @@ func (b *backend) credsReadHandler(ctx context.Context, req *logical.Request, d
 		return nil, err
 	}
-	username := fmt.Sprintf("vault_%s_%s_%s_%d", name, req.DisplayName, userUUID, time.Now().UnixNano())
+	username := fmt.Sprintf("%s_%s", role.UserPrefix, userUUID)
```
userUUID might not be unique, e.g., did you test same user requesting two emphemeral creds with overlapping lifetime?

michaelw replied:
also, should we keep requestor name in there for tracking? (at least unless explicitly overridden)

abhide replied:
A new UUID is generated for userUUID for each new user. Check 3 lines above

abhide replied:
Added req.DisplayName to username

Commit:
Fix `make dev`

Accidentally, delete `go install` from `make build` target;
while moving to go modules.

michaelw reviewed (approved):
we might want to document that `defaultUserPrefix` is a special value now (or not make it special)
